WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift

```diff
@@ -53,23 +54,39 @@ final class PointOfSaleDashboardViewModel: ObservableObject {
     /// If the merchant goes back to the product selection screen and makes changes, this should be updated when they return to the checkout.
     @Published private var order: POSOrder?
     @Published private(set) var isSyncingOrder: Bool = false
-    private let orderService: POSOrderServiceProtocol
+    @Published private(set) var isSyncingItems: Bool = true
 
+    private let orderService: POSOrderServiceProtocol
+    private let itemProvider: POSItemProvider
     private let cardPresentPaymentService: CardPresentPaymentFacade
 
     private let currencyFormatter = CurrencyFormatter(currencySettings: ServiceLocator.currencySettings)
 
-    init(items: [POSItem],
+    init(itemProvider: POSItemProvider,
          cardPresentPaymentService: CardPresentPaymentFacade,
          orderService: POSOrderServiceProtocol) {
-        self.items = items
+        self.itemProvider = itemProvider
         self.cardPresentPaymentService = cardPresentPaymentService
         self.cardReaderConnectionViewModel = CardReaderConnectionViewModel(cardPresentPayment: cardPresentPaymentService)
         self.orderService = orderService
+
+        populatePointOfSaleItems()
+
         observeCardPresentPaymentEvents()
         observeItemsInCartForCartTotal()
     }
 
+    private func populatePointOfSaleItems() {
+        Task { @MainActor in
+            do {
+                items = try await itemProvider.providePointOfSaleItemsFromNetwork()
```

Comment:
> You could even make it a var, and have the items be fetched when it's created, adding a refresh function as well. Better to avoid having a var do the fetch though, not least because it's async.

Thanks for the tip! I'll check this when adding the pull-to-refresh here https://github.com/woocommerce/woocommerce-ios/issues/12837